gitdiff/patch_header.go

```diff
@@ -369,3 +373,61 @@ func parseHeaderMail(mailLine string, r io.Reader) (*PatchHeader, error) {
 
 	return h, nil
 }
+
+func cleanupSubject(s string) string {
+	r := []rune(s)
+
+	// This is meant to be compatible with
+	// https://github.com/git/git/blob/master/mailinfo.c:cleanup_subject().
+	// If compatibility with `git am` drifts, go there to see if there
+	// are any updates.
+
+	var at int
+	for at = 0; at < len(r); {
```

Comment:
style: I'd either remove the `a = 0` as redundant, or if you think it improves clarity, write the definition on L385 as `at := 0`

Reply:
Ack